Code/GraphMol/SubstructLibrary/Wrap/SubstructLibraryWrap.cpp

```diff
@@ -0,0 +1,182 @@
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+#include <RDBoost/python.h>
+#include <RDBoost/Wrap.h>
+#include <GraphMol/RDKitBase.h>
+
+#include <GraphMol/SubstructLibrary/SubstructLibrary.h>
+
+namespace python = boost::python;
+
+namespace RDKit {
+
+const char * MolHolderBaseDoc = "";
+const char * MolHolderDoc = "";
+const char * CachedMolHolderDoc = "";
+const char * CachedSmilesMolHolderDoc = "";
+const char * PatternHolderDoc = "";
+const char * SubstructLibraryDoc = "";
+
+struct substructlibrary_wrapper {
+    static void wrap() {
+      // n.b. there can only be one of these in all wrappings
+      //python::class_<std::vector<unsigned int> >("UIntVect").def(
+      //  python::vector_indexing_suite<std::vector<unsigned int>, true>());
+
+      python::class_<MolHolderBase, boost::noncopyable>(
+          "MolHolderBase", "", python::no_init)
+          
+          .def("AddMol", &MolHolderBase::addMol,
+               "Adds molecle to the molecule holder")
+          .def("GetMol", &MolHolderBase::getMol,
+               "Returns a particular molecule in the molecule holder\n\n"
+               "  ARGUMENTS:\n"
+               "    - idx: which molecule to return\n\n"
+               "  NOTE: molecule indices start at 0\n")
+          .def("__len__", &MolHolderBase::size);
+
+      python::class_<MolHolder, MolHolder*, MolHolder&, python::bases<MolHolderBase> >(
+          "MolHolder", MolHolderDoc, python::init<>());
+
+      python::class_<CachedMolHolder, python::bases<MolHolderBase > >(
+          "CachedMolHolder", CachedMolHolderDoc, python::init<>());
+
+      python::class_<CachedSmilesMolHolder, python::bases<MolHolderBase > >(
+          "CachedSmilesMolHolder", CachedSmilesMolHolderDoc, python::init<>());
+
+      python::class_<FPHolderBase, boost::noncopyable>(
+          "FPHolderBase", "", python::no_init)
+          .def("AddMol", &FPHolderBase::addMol,
+               "Adds molecle to the fingerprint database")
+          .def("PassesFilter", &FPHolderBase::passesFilter);
+      
+      python::class_<PatternHolder, PatternHolder*, PatternHolder&, python::bases<FPHolderBase> >(
+          "PatternHolder", PatternHolderDoc, python::init<>());
+      
+      python::class_<SubstructLibrary, SubstructLibrary *,
+                     const SubstructLibrary *>(
+                         "SubstructLibrary", SubstructLibraryDoc, python::init<>())
+          .def(python::init<MolHolderBase *>())          
```

Comment:
you can use shared pointers (probably the easiest) or you can tie the lifetime of things together:
http://www.boost.org/doc/libs/1_56_0/libs/python/doc/v2/with_custodian_and_ward.html